flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkOperatorConfiguration.java

```diff
@@ -185,6 +186,9 @@ public static FlinkOperatorConfiguration fromConfiguration(Configuration operato
         boolean savepointOnDeletion =
                 operatorConfig.get(KubernetesOperatorConfigOptions.SAVEPOINT_ON_DELETION);
 
+        boolean drainJobOnSavepointDeletion =
+                operatorConfig.get(KubernetesOperatorConfigOptions.DRAIN_ON_SAVEPOINT_DELETION);
```

Comment:
Similar to this fix PR: https://github.com/apache/flink-kubernetes-operator/pull/659

These configs should not be part of the `FlinkOperatorConfiguration` which will prevent it for setting it for a per-resource level. It should be accessed from the observeConfig directly.

Reply:
Got it!
Updated the PR to reflect these changes.